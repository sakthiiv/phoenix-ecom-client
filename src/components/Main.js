import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Home from './Home'
import Admin from './Admin'
import CreateCategory from './CreateCategory'
import CreateProduct from './CreateProduct'

const Main = () => (
  <main>
    <Switch>
      <Route exact={true} path='/' component={Home}/>
      <Route exact={true} path='/admin' component={Admin}/>
      <Route exact={true} path='/create-category' component={CreateCategory}/>
      <Route exact={true} path='/create-product' component={CreateProduct}/>
    </Switch>
  </main>
)

export default Main
