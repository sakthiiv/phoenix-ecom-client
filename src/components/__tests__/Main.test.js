import React from 'react';
import { shallow } from 'enzyme';
import Main from '../Main';
describe('Main', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Main />);
      expect(wrapper).toMatchSnapshot();
      // On the first run of this test, Jest will generate a snapshot file automatically.
    });
});