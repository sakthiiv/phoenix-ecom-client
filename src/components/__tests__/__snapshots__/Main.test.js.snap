// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="container"
>
  <main>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/userproductList"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/create-category"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/create-product"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category-list"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/product-list"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/product-detail"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/shopping-cart"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/search"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/checkout"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
</div>
`;
